website/addons/dmptool/model.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+
+
+from modularodm import fields
+
+from framework.exceptions import HTTPError
+
+from website.addons.base import (
+    AddonOAuthNodeSettingsBase, AddonOAuthUserSettingsBase, exceptions,
+)
+from website.addons.base import StorageAddonBase
+
+from website.addons.dmptool.serializer import DmptoolSerializer
+
+ROOT_FOLDER_NAME = '/'
+
+
+class DmptoolProvider(object):
+    """An alternative to `ExternalProvider` not tied to OAuth"""
+
+    name = 'Dmptool'
+    short_name = 'dmptool'
+    serializer = DmptoolSerializer
+
+    def __init__(self, account=None):
+        super(DmptoolProvider, self).__init__()
+        # provide an unauthenticated session by default
+        self.account = account
+
+    def __repr__(self):
+        return '<{name}: {status}>'.format(
+            name=self.__class__.__name__,
+            status=self.account.provider_id if self.account else 'anonymous'
+        )
+
+class AddonDmptoolUserSettings(AddonOAuthUserSettingsBase):
+    oauth_provider = DmptoolProvider
+    serializer = DmptoolSerializer
+
+class AddonDmptoolNodeSettings(StorageAddonBase, AddonOAuthNodeSettingsBase):
+    oauth_provider = DmptoolProvider
+    serializer = DmptoolSerializer
+
+    folder_id = fields.StringField(default=ROOT_FOLDER_NAME)
+    folder_name = fields.StringField(default=ROOT_FOLDER_NAME)
+    folder_path = fields.StringField(default=ROOT_FOLDER_NAME)
```

Comment:
Yep, you do need all three, but looking down at `def set_folder()`, you set both `folder_path` and `folder_name` to `ROOT_FOLDER_NAME`, but `folder_id` to `str(folder_id)`.  I think `folder_id` and `folder_path` can both be `str(folder_id)`, but `folder_name` should be `/ (Full DMPtool)`